### ISSUE TITLE
Using old cached target logits

### ISSUE THREAD

**Issue by henrikfo** [aidotse/LeakPro#346]
# Issue
## Problem Description
After retraining a target model, the code will check for any existing a cache logits. If there are logits cached already the logits from the new target model will not be cached and the old ones will be used. This might lead to unexpected erroneous results.

## Expected Behavior
The logits from the retrained target model should be used 

## What Needs to be Done
Some check that make sure that a new target model caches the new logits

## How Can It Be Tested or Reproduced
Run any leakpro example notebook. Then rerun by changing a parameter in target_config.yaml (e.g. +1/-1 epochs) and then check the logs when running leakpro.run_audit(). It will say "Logits already cached at leakpro_output/attack_cache/target_logits.npy"
